sunpy/lightcurve/sources/soho.py

```diff
@@ -0,0 +1,158 @@
+        #     ax2.set_yscale("log")
+        #     ax2.set_ylim(1e-9, 1e-2)
+        #     ax2.set_yticks((1e-9, 1e-8, 1e-7, 1e-6, 1e-5, 1e-4, 1e-3, 1e-2))
+        #     ax2.set_yticklabels((' ', 'A', 'B', 'C', 'M', 'X', ' '))
+        
+        #     axes.yaxis.grid(True, 'major')
+        #     axes.xaxis.grid(False, 'major')
+        #     axes.legend()
+        
+        #     # @todo: display better tick labels for date range (e.g. 06/01 - 06/05)
+        #     formatter = matplotlib.dates.DateFormatter('%H:%M')
+        #     axes.xaxis.set_major_formatter(formatter)
+        
+        #     axes.fmt_xdata = matplotlib.dates.DateFormatter('%H:%M')
+        #     figure.autofmt_xdate()
+        #     figure.show()
+        
+        #     return figure
+
+
+        @staticmethod
+        def _parse_txt(filepath):
+                """
+                Parses a SOHO/ERNE file from
+                http://srl.utu.fi/erne_data/carrot/carrota.html
+                http://srl.utu.fi/erne_data/carrot/carrotp.html
+                
+                and returns header as a list and ERNE data as pandas dataframe
+                """
+                
+                #Reading in Data along with header
+                data = ascii.read(filepath, delimiter = "\s", data_start = 2 ) 
+                
+                #header is read along with data i.e the first row in data is the header
+                header = data.colnames
+                #So extracting the first row elements and forming header list that contains the energy bins
+                header = [data[0][key] for key in data.colnames]
+                #and now excluding the first line to have only the data in data variable
+                data = data[1:]
+                
+                data_modify = []
+                
+                #Storing data columns in recognizable variables
+                year_col = data['col1']
+                month_col = data['col2']
+                date_col = data['col3']
+                begin_time_col = data['col4']
+                end_time_col = data['col5']
+                
+                #Converting separate datetime element into a single datetime.datetime column
+                for i in range(len(data)):
+                    
+                        #Combining separate datetime elements into single datetime value
+                        #start time
+                        date1 = datetime.strptime(year_col[i] + '-' + month_col[i] + '-' + date_col[i] + '/' + begin_time_col[i][:2] + ':' +begin_time_col[i][2:],"%y-%m-%d/%H:%M")
+                        #end time
+                        date2 = datetime.strptime(year_col[i] + '-' + month_col[i] + '-' + date_col[i] + '/' + end_time_col[i][:2] + ':' + end_time_col[i][2:], "%y-%m-%d/%H:%M")
+                        #Appending the start and end time as sunpy.time TimeRange in a separate list
+                        data_modify.append(TimeRange(date1, date2))
+                
```

Comment:
didn't get you here !! First few lines of  sample data is as follows: full
file : here <http://www.srl.utu.fi/erne_data/carrot/1907/cr1907p.txt>

Date          begin end      Intensities [1/(cm^2*sr*s*MeV/n)] by energy
channels [MeV/n]
yy mm dd hhmm hhmm  1.8-3.3   3.3-6.4   6.4-12.7  13.5-25.8 25.8-50.7

96 03 11 0000 0200  8.75E-04  4.90E-05 -1.00E+00  6.40E-05  3.71E-06
96 03 11 0200 0400  4.34E-04  4.88E-05  4.82E-05  5.47E-05  4.55E-06
96 03 11 0400 0600  1.08E-03 -1.00E+00  1.44E-04  6.60E-05  3.90E-06
96 03 11 0600 0800  1.52E-03  4.86E-05  4.80E-05  5.10E-05  3.69E-06
96 03 11 0800 1000  6.50E-04 -1.00E+00  9.63E-05  6.02E-05  2.82E-06
96 03 11 1000 1200  6.50E-04  4.88E-05  4.80E-05  6.20E-05  3.03E-06
96 03 11 1200 1400  2.17E-03 -1.00E+00  9.61E-05  5.14E-05  1.30E-06
96 03 11 1400 1600  1.10E-03 -1.00E+00  4.89E-05  5.86E-05  3.97E-06
96 03 11 1600 1800  6.55E-04  9.78E-05  9.70E-05  5.88E-05  3.28E-06
96 03 11 1800 2000  1.16E-03 -1.00E+00  2.58E-05  6.13E-05  4.88E-06
96 03 11 2000 2200  1.12E-03 -1.00E+00  2.49E-05  6.07E-05  3.59E-06
96 03 11 2200 0000  1.08E-03  4.86E-05  7.21E-05  5.98E-05  3.47E-06
96 03 12 0000 0200  1.30E-03  4.88E-05  2.40E-05  5.47E-05  3.25E-06

And I didnt quite get you why would the time range be negative ? the date
changes i.e day changes !!
ᐧ

On Thu, Jul 2, 2015 at 3:24 PM, dpshelio <notifications@github.com> wrote:

> In sunpy/lightcurve/sources/soho.py
> <https://github.com/sunpy/sunpy/pull/1476#discussion_r33762883>:
>
> > +                month_col = data['col2']
> > +                date_col = data['col3']
> > +                begin_time_col = data['col4']
> > +                end_time_col = data['col5']
> > +
> > +                #Converting separate datetime element into a single datetime.datetime column
> > +                for i in range(len(data)):
> > +
> > +                        #Combining separate datetime elements into single datetime value
> > +                        #start time
> > +                        date1 = datetime.strptime(year_col[i] + '-' + month_col[i] + '-' + date_col[i] + '/' + begin_time_col[i][:2] + ':' +begin_time_col[i][2:],"%y-%m-%d/%H:%M")
> > +                        #end time
> > +                        date2 = datetime.strptime(year_col[i] + '-' + month_col[i] + '-' + date_col[i] + '/' + end_time_col[i][:2] + ':' + end_time_col[i][2:], "%y-%m-%d/%H:%M")
> > +                        #Appending the start and end time as sunpy.time TimeRange in a separate list
> > +                        data_modify.append(TimeRange(date1, date2))
> > +
>
> Also... Is it there just a file per day? what happens if one measurement
> ends in the following day? say, the end time is 0003 as it's above the
> Timerange would be negative...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sunpy/sunpy/pull/1476/files#r33762883>.
>